usecases/api_key_usecase.go

```diff
@@ -0,0 +1,120 @@
+package usecases
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/checkmarble/marble-backend/models"
+	"github.com/checkmarble/marble-backend/repositories"
+	"github.com/checkmarble/marble-backend/usecases/analytics"
+	"github.com/checkmarble/marble-backend/usecases/transaction"
+	"github.com/google/uuid"
+)
+
+type ApiKeyRepository interface {
+	GetApiKeyById(ctx context.Context, tx repositories.Transaction, apiKeyId string) (models.ApiKey, error)
+	ListApiKeys(ctx context.Context, tx repositories.Transaction, organizationId string) ([]models.ApiKey, error)
+	CreateApiKey(ctx context.Context, tx repositories.Transaction, apiKey models.CreateApiKey) error
+	SoftDeleteApiKey(ctx context.Context, tx repositories.Transaction, apiKeyId string) error
+}
+
+type EnforceSecurityApiKey interface {
+	CreateApiKey(organizationId string) error
+	DeleteApiKey(apiKey models.ApiKey) error
+	ListApiKeys() error
+}
+
+type ApiKeyUseCase struct {
+	transactionFactory      transaction.TransactionFactory
+	organizationIdOfContext func() (string, error)
+	enforceSecurity         EnforceSecurityApiKey
+	apiKeyRepository        ApiKeyRepository
+}
+
+func (usecase *ApiKeyUseCase) ListApiKeys(ctx context.Context) ([]models.ApiKey, error) {
+	if err := usecase.enforceSecurity.ListApiKeys(); err != nil {
+		return []models.ApiKey{}, err
+	}
+
+	organizationId, err := usecase.organizationIdOfContext()
+	if err != nil {
+		return []models.ApiKey{}, err
+	}
+
+	return usecase.apiKeyRepository.ListApiKeys(ctx, nil, organizationId)
```

Comment:
Here, I'd like to do: for every api key in output, validate that its org id corresponds to the input org id (see for instance how it's done in cases)